src/models/project.js

```diff
@@ -0,0 +1,118 @@
+module.exports = (sequelize, DataTypes) => {
+  const Project = sequelize.define('Project', {
+    id: {
+      allowNull: false,
+      primaryKey: true,
+      type: DataTypes.UUID,
+      defaultValue: DataTypes.UUIDV4
+    },
+    teamId: {
+      type: DataTypes.UUID,
+      allowNull: false,
+      onDelete: 'CASCADE',
+      references: {
+        model: 'Teams',
+        key: 'id',
+        as: 'teamId',
+      },
+      validate: {
+        notEmpty: {
+          args: true,
+          msg: 'The teamId field is required'
+        },
+        async findTeam(value) {
+          const team = await sequelize.models.Team.findById(value);
+          if (!team) {
+            throw new Error(`A team with the id of ${value} does not exist`);
+          }
+        },
+      },
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notEmpty: {
+          args: true,
+          msg: 'The name field is required'
+        },
+        async exist(value) {
+          const existingProject = await Project.findOne({
+            where: { teamId: this.teamId, name: value }
+          });
+          if (existingProject) {
+            throw new
+            Error('Project with the same name already exists in this team.');
```

Comment:
@Chieze-Franklin This is the check for existing project names. But is it ok to throw this error if a user tries to update a project with the same name. i.e update a project with the name of  `cfh` back to `cfh`?

Reply:
No I think we should silently ignore a change  like from `cfh` to `cfh`. We shouldn't throw an error.